integration/wait.go

```diff
@@ -66,3 +67,26 @@ func WaitForObject(
 		return checkFn(object)
 	})
 }
+
+// RetryUntilNoError retries a function for a specified duration until it returns no error
+func RetryUntilNoError(retryFor, sleep time.Duration, f func() error) (err error) {
+	var e error
+	start := time.Now()
+
+	log.Printf("retrying function every %s, for %s", sleep, retryFor)
+	for {
+		e = f()
+		if e == nil {
+			return nil
+		}
+
+		now := time.Now()
+		if now.Sub(start) >= retryFor {
+			log.Println("retry deadline reached")
```

Comment:
You need to hand in `t *testing.T` and log on it, or the logout is either hidden or might be out of order when running tests in parallel.